### PR TITLE
Removed capture

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/expr/tests.py [hunk 1]
```diff
@@ -1005,13 +1005,13 @@ def test_array_methods(self):
         self.assertEqual(hl.eval_expr(hl.flatmap(lambda x: hl.range(0, x), [1, 2, 3])), [0, 0, 1, 0, 1, 2])
 
     def test_bool_r_ops(self):
-        self.assertTrue(hl.eval_expr(hl.capture(True) & True))
-        self.assertTrue(hl.eval_expr(True & hl.capture(True)))
-        self.assertTrue(hl.eval_expr(hl.capture(False) | True))
-        self.assertTrue(hl.eval_expr(True | hl.capture(False)))
+        self.assertTrue(hl.eval_expr(hl.lit(True) & True))
```
Comment (cseed) [inline python/hail/expr/tests.py]: Do we prefer hl.bool(True) in this case when we clearly know the type?

File: python/hail/expr/functions.py [hunk 2]
```diff
@@ -78,9 +78,14 @@ def lit(x, dtype=None):
...
     :class:`.Expression`
     """
+    imputed = dtype is None
     if dtype is None:
         dtype = impute_type(x)
-    dtype._typecheck(x)
+    try:
+        dtype._typecheck(x)
+    except TypeError as e:
+        raise TypeError("'lit': object did not match the {} type '{}'"
```
Comment (cseed) [inline python/hail/expr/functions.py]: If _typecheck fails in the imputed case, isn't there a bug in impute_type?  I think you can write:
```
if dtype:
  try:
    dtype._typecheck(x)
  except TypeError as e:
    raise ...
else:
  dtype = impute_type(x)
...
```

Comment (cseed) [reply python/hail/expr/functions.py]: Maybe with an assertion in the second case.

File: python/hail/expr/tests.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/expr/tests.py]: I'm not sure. I just did find/replace here. Happy to change it if you think that's appropriate

Comment (cseed) [reply python/hail/expr/tests.py]: I don't have a strong opinion.  There were a few cases of `hl.lit([1, 2, 3])` and `hl.array([1, 2, 3])` and I thought I'd raise the question.  I'm happy with it as-is.

File: python/hail/expr/functions.py [hunk 3]
```diff
@@ -2762,6 +2737,32 @@ def struct(**kwargs):
     """
     return to_expr(Struct(**kwargs))
 
+def tuple(args):
```
Comment (jigold) [inline python/hail/expr/functions.py]: Should this be varargs?

Comment (cseed) [reply python/hail/expr/functions.py]: I'm with @tpoterba:

```
>>> tuple([1, 2, 3])
(1, 2, 3)
>>> tuple(1, 2, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: tuple() takes at most 1 argument (3 given)
```

Comment (jigold) [reply python/hail/expr/functions.py]: Ok. I think the docs need to be clearer for what parameters are accepted. The builtin accepts any iterable object in Python. Looks like lists, sets, tuples all behave as expected. Dictionaries return a tuple of the keys.

Comment (tpoterba) [reply python/hail/expr/functions.py]: yeah, the docs are actually just wrong as written.

**Comment by cseed** [comment]
Hail is no longer :lit: :sad:

**Pull request merged**